modules/common/src/bugsnag/bugsnag.filter.ts

```diff
@@ -0,0 +1,46 @@
+import {Catch, ExceptionFilter, HttpException, HttpStatus} from '@nestjs/common';
+import {ArgumentsHost} from '@nestjs/common/interfaces/features/arguments-host.interface';
+import {Response} from 'express';
+import * as httpContext from 'express-http-context';
+import * as jsonStringifySafe from 'json-stringify-safe';
+import {REQUEST_UNIQUE_ID_KEY} from '../constants';
+import {clearBreadcrumbs, getBreadcrumbs} from './breadcrumbs';
+import {BugsnagClient} from './bugsnag.client';
+import {BugsnagSeverity} from './interfaces';
+
+@Catch()
+export class BugsnagErrorFilter implements ExceptionFilter {
+    constructor (
+        private readonly bugsnagClient: BugsnagClient,
+        private readonly loggerMethod?: Function,
+    ) {}
+
+    public catch (err: any, host: ArgumentsHost): Response {
+        const res: Response = host.switchToHttp().getResponse();
+        const status: number = err instanceof HttpException ? err.getStatus() : HttpStatus.INTERNAL_SERVER_ERROR;
+        const exception: Error = err instanceof Error ? err : new Error(err);
+
+        // tslint:disable-next-line:no-unbound-method
+        (this.loggerMethod || console.error)(
+            // Stringify when err message is an object to avoid [object Object] only
+            `>> status: "${status}" - "${jsonStringifySafe(exception)}" - "${jsonStringifySafe(exception.stack)}"`,
+        );
+
+        this.bugsnagClient.notifyWithMetadata(
+            exception,
+            {
+                severity: BugsnagSeverity.ERROR,
+                breadcrumbs: getBreadcrumbs(),
+                metadata: {
+                    uniqueId: (httpContext.get(REQUEST_UNIQUE_ID_KEY) || 'unknown'),
```

Comment:
I just saw here that the fallback for uniqueId is "'unkown'" while in the logger `getUniqueKey` it's "''" an empty string. Wouldn't it be good to have a consistent fallback?